WebDriverAgentLib/Utilities/FBSearchPredicatesFormatter.m

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "FBSearchPredicatesFormatter.h"
+#import "FBElement.h"
+#import "FBElementUtils.h"
+
+@implementation NSPredicate (ExtractComparisons)
+
+- (NSPredicate *)predicateByChangingComparisonsWithBlock:(NSPredicate *(^)(NSComparisonPredicate *))block {
+  if ([self isKindOfClass: [NSCompoundPredicate class]]) {
+    NSCompoundPredicate *compPred = (NSCompoundPredicate *)self;
+    NSMutableArray *predicates = [NSMutableArray array];
+    for (NSPredicate *predicate in [compPred subpredicates]) {
+      NSPredicate *newPredicate = [predicate predicateByChangingComparisonsWithBlock: block];
+      if (newPredicate != nil) {
+        [predicates addObject: newPredicate];
+      }
+    }
+    return [[NSCompoundPredicate alloc] initWithType: compPred.compoundPredicateType
+                                        subpredicates: predicates];
+  }
+  if ([self isKindOfClass: [NSComparisonPredicate class]]) {
+    return block((NSComparisonPredicate *)self);
+  }
+  return self;
+}
+
+@end
+
+NSString *const FBUnknownPredicateKeyException = @"FBUnknownPredicateKeyException";
+
+@implementation FBSearchPredicatesFormatter
+
++ (NSExpression *)formatExpression:(NSExpression *)input validPropertyNames:(NSArray<NSString *> *)validPropertyNames
```

Comment:
It would be a good idea to move that method to `NSExpression` category as constructor and write tests for it.